bgpd/bgpd.c

```diff
@@ -6573,6 +6574,192 @@ int peer_unsuppress_map_unset(struct peer *peer, afi_t afi, safi_t safi)
 	return 0;
 }
 
+/* Set advertise-map to the peer but do not process peer route updates here.  *
+ * Hold filter changes until the conditional routes polling thread is called  *
+ * AS we need to advertise/withdraw prefixes (in advertise-map) based on the  *
+ * condition (exist-map/non-exist-map) and routes(specified in condition-map) *
+ * in BGP table. So do not call peer_on_policy_change() here, only create     *
+ * polling timer thread, update filters and increment condition_filter_count.
+ */
+int peer_advertise_map_set(struct peer *peer, afi_t afi, safi_t safi,
+			   const char *advertise_name,
+			   struct route_map *advertise_map, bool condition,
+			   const char *condition_name,
+			   struct route_map *condition_map)
+{
+	bool filter_exists = false;
+	struct peer *member;
+	struct bgp_filter *filter;
+	struct listnode *node, *nnode;
+
+	/* Set configuration on peer. */
+	filter = &peer->filter[afi][safi];
+
+	if (filter->advmap.aname) {
+		/* advertise-map filter is already configured on this peer */
+		filter_exists = true;
+
+		XFREE(MTYPE_BGP_FILTER_NAME, filter->advmap.aname);
+		XFREE(MTYPE_BGP_FILTER_NAME, filter->advmap.cname);
+		filter->advmap.condition = CONDITION_NON_EXIST;
+	}
+
+	route_map_counter_decrement(filter->advmap.amap);
+	filter->advmap.aname = XSTRDUP(MTYPE_BGP_FILTER_NAME, advertise_name);
+	filter->advmap.cname = XSTRDUP(MTYPE_BGP_FILTER_NAME, condition_name);
+	filter->advmap.amap = advertise_map;
+	filter->advmap.cmap = condition_map;
+	filter->advmap.condition = condition;
+	route_map_counter_increment(advertise_map);
+	peer->advmap_info[afi][safi].config_change = true;
+
+	/* Check if handling a regular peer. */
+	if (!CHECK_FLAG(peer->sflags, PEER_STATUS_GROUP)) {
+		/* Set override-flag and process peer route updates. */
+		SET_FLAG(peer->filter_override[afi][safi][RMAP_OUT],
+			 PEER_FT_ADVERTISE_MAP);
+
+		/* Hold peer_on_policy_change() until timer thread is called */
```

Comment:
something is missing here or lonely comment? :)

Reply:
no just a comment. :)